### PR TITLE
fix(formatter): add break to ts constructor parameters with modifiers

### LINKED ISSUE #849: 🐛 Break on TypeScript parameter properties
### Environment information

```block
Playground
```

### What happened?

Biome should break lines when at least one parameter property is present.
See the related prettier PR [prettier#14402](https://github.com/prettier/prettier/pull/14402).

[Playground link](https://biomejs.dev/playground?trailingComma=es5&lintRules=all&code=YwBsAGEAcwBzACAATQB5AEMAbABhAHMAcwAgAHsACgAgACAAYwBvAG4AcwB0AHIAdQBjAHQAbwByACgACgAgACAAIAAgAGIAOgAgAHMAdAByAGkAbgBnACwACgAgACAAIAAgAHAAcgBvAHQAZQBjAHQAZQBkACAAeAA6ACAAbgB1AG0AYgBlAHIALAAKACAAIAAgACAAYQA6ACAAcwB0AHIAaQBuAGcALAAKACAAIAApACAAewB9AAoAfQAKAA%3D%3D)

### Expected result

The following code:

```ts
class MyClass {
  constructor(a: string, protected x: number, b: string) {}
}
```

should be formatted as Prettier:

```ts
class MyClass {
	constructor(
		a: string,
		protected x: number,
		b: string
	) {}
}
```

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

### PR DESCRIPTION
**Pull request by TaKO8Ki**
## Summary

closes #849

I've added break to TypeScript constructor parameters with modifiers.

## Test Plan

I have added a test case, All others test should pass.

### CODE REVIEW

File: crates/biome_js_formatter/src/js/lists/parameter_list.rs [hunk 1]
```diff
@@ -67,12 +67,22 @@ impl Format<JsFormatContext> for FormatJsAnyParameterList<'_> {
                     FormatTrailingComma::All.trailing_separator(f.options())
                 };
 
+                let has_modifiers = self.list.iter().any(|node| {
+                    matches!(node, Ok(AnyParameter::AnyJsConstructorParameter(
+                        AnyJsConstructorParameter::TsPropertyParameter(parameter),
+                    )) if !parameter.modifiers().is_empty())
```
Comment (Conaclos) [inline crates/biome_js_formatter/src/js/lists/parameter_list.rs:73]: Do we really need t ocheck that the list is not empty? It is already guaren,ted that it is a parameter property?
```suggestion
                    ))
```

**Comment by Conaclos** [comment]
Looks good to me! I just left a comment :)

File: crates/biome_js_formatter/src/js/lists/parameter_list.rs (hunk 1 above)
Comment (TaKO8Ki) [reply crates/biome_js_formatter/src/js/lists/parameter_list.rs:73]: Indeed, you might be right about that. I will remove this condition.

Comment (TaKO8Ki) [reply crates/biome_js_formatter/src/js/lists/parameter_list.rs]: I have removed it.

**Review by Conaclos — APPROVED** [review]

**Pull request merged**